src/main/java/seedu/duke/storage/Storage.java

```diff
@@ -1,4 +1,35 @@
 package seedu.duke.storage;
 
-public class Storage {
+import java.io.File;
+import java.io.IOException;
+
+public abstract class Storage {
+    File file;
+
+    public Storage(String dir) {
+        createFile(dir);
+    }
+
+    protected void createFile(String dir) {
+        try {
+            file = new File(dir);
+            file.getParentFile().mkdirs();
+            if (file.createNewFile()) {
```

Comment:
What does file.createNewFile() return?
Shouldn't this be useing file.exists() function?
If exist, it will create a new file.
Else, file already exists?